packages/itwin/grouping-mapping-widget/src/widget/components/GroupAction.tsx

```diff
@@ -281,64 +345,53 @@ const GroupAction = ({
           <RadioTileGroup
-            className="gmw-radio-group-tile"
+            className='gmw-radio-group-tile'
             required>
-            <RadioTile
-              name={"groupby"}
-              icon={<SvgCursor />}
-              onChange={changeGroupByType}
-              defaultChecked
-              value={"Selection"}
-              label={"Selection"}
-              disabled={isLoading || isRendering}
-            />
-            <RadioTile
-              icon={<SvgSearch />}
-              name={"groupby"}
-              onChange={changeGroupByType}
-              value={"Query Keywords"}
-              label={"Query Keywords"}
-              disabled={isLoading || isRendering}
-            />
-          </RadioTileGroup>
-          {queryGenerationType === "Selection" ?
-            <GroupQueryBuilderContext.Provider
-              value={{
-                currentPropertyList,
-                setCurrentPropertyList,
-                query,
-                setQuery,
-                queryBuilder,
-                setQueryBuilder,
-                isLoading,
-                isRendering,
-                resetView,
-              }}
-            >
-              <GroupQueryBuilderContainer />
-            </GroupQueryBuilderContext.Provider> :
-            <div className="gmw-search-form">
-              <Text>Generate a query by keywords. Keywords wrapped in double quotes will be considered a required criteria.</Text>
-              <LabeledTextarea
-                label="Query Keywords"
-                required
-                value={searchInput}
-                onChange={(event) => setSearchInput(event.target.value)}
+            {(uiProviders === undefined || uiProviders.length === 0) && (
+              <>
+                <RadioTile
+                  name={"groupby"}
+                  icon={<SvgCursor />}
+                  onChange={changeGroupByType}
+                  value={"Selection"}
+                  label={"Selection"}
+                  disabled={isLoading || isRendering}
+                  checked={localQueryGenerationType === "Selection"}
+                />
+                <RadioTile
+                  icon={<SvgSearch />}
+                  name={"groupby"}
+                  onChange={changeGroupByType}
+                  value={"Search"}
+                  label={"Query Keywords"}
+                  disabled={isLoading || isRendering}
+                  checked={localQueryGenerationType === "Search"}
+                />
+                <RadioTile
+                  icon={<SvgDraw />}
+                  name={"groupby"}
+                  onChange={changeGroupByType}
+                  value={"Manual"}
+                  label={"Manual Query"}
+                  disabled={isLoading || isRendering}
+                  checked={localQueryGenerationType === "Manual"}
+                />
+              </>
+            )}
+            {uiProviders?.map((ext) => (
```

Comment:
`uiProviders` will never be null or undefined, so you can drop the optional chaining operator.